sdk/protocol-plugins/src/plugins/aave-v3/builders/AaveV3DepositBorrowActionBuilder.ts

```diff
@@ -3,91 +3,100 @@ import {
-    arguments: {
-      approvalAmount: getValueFromReference(step.inputs.depositAmount),
-      delegate: aaveV3LendingPoolAddress,
-      sumAmounts: false,
-    },
-    connectedInputs: {
-      depositAmount: 'approvalAmount',
-    },
-    connectedOutputs: {},
-  })
-
-  context.addActionCall({
-    step: params.step,
-    action: new AaveV3DepositAction(),
-    arguments: {
-      depositAmount: getValueFromReference(step.inputs.depositAmount),
-      sumAmounts: false,
-      setAsCollateral: true,
-    },
-    connectedInputs: {
-      depositAmount: 'amountToDeposit',
-    },
-    connectedOutputs: {
-      depositAmount: 'depositedAmount',
-    },
-  })
-
-  const borrowAmount = getValueFromReference(step.inputs.borrowAmount)
+    const aaveV3LendingPoolAddress = await getContractAddress({
+      addressBookManager,
+      chainInfo: user.chainInfo,
+      contractName: 'AaveV3LendingPool',
+    })
 
-  if (!borrowAmount.toBN().isZero()) {
     context.addActionCall({
       step: step,
-      action: new AaveV3BorrowAction(),
+      action: new SetApprovalAction(),
+      arguments: {
+        approvalAmount: getValueFromReference(step.inputs.depositAmount),
+        delegate: aaveV3LendingPoolAddress,
+        sumAmounts: false,
+      },
+      connectedInputs: {
+        depositAmount: 'approvalAmount',
+      },
+      connectedOutputs: {},
+    })
+
+    context.addActionCall({
+      step: params.step,
+      action: new AaveV3DepositAction(),
       arguments: {
-        borrowAmount: borrowAmount,
-        borrowTo: await getBorrowTargetAddress(params),
+        depositAmount: getValueFromReference(step.inputs.depositAmount),
+        sumAmounts: false,
+        setAsCollateral: true,
+      },
+      connectedInputs: {
+        depositAmount: 'amountToDeposit',
       },
-      connectedInputs: {},
       connectedOutputs: {
-        borrowAmount: 'borrowedAmount',
+        depositAmount: 'depositedAmount',
       },
     })
+
+    const borrowAmount = getValueFromReference(step.inputs.borrowAmount)
+
+    if (!borrowAmount.toBN().isZero()) {
+      context.addActionCall({
+        step: step,
+        action: new AaveV3BorrowAction(),
+        arguments: {
+          borrowAmount: borrowAmount,
+          borrowTo: await this._getBorrowTargetAddress(params),
+        },
+        connectedInputs: {},
+        connectedOutputs: {
+          borrowAmount: 'borrowedAmount',
+        },
+      })
+    }
+  }
+
+  /**
+   * Resolves the target address for the borrow action based on the borrow target type
+   * @param params The parameters for the action builder
+   * @returns The address of the target contract
+   */
+  private async _getBorrowTargetAddress(
+    params: ActionBuilderParams<steps.DepositBorrowStep>,
+  ): Promise<IAddress> {
+    const { user, step, positionsManager, addressBookManager } = params
+    if (step.inputs.borrowTargetType === TokenTransferTargetType.PositionsManager) {
```

Comment:
maybe check explicitely for other `TokenTransferTargetType`s and throw if it's not known

Reply:
I'll add a switch case for this so I can throw an error on `default`